__tests__/middleware-test.js

```diff
@@ -0,0 +1,158 @@
+jest.mock('@agoric/casting', () => {
+    return {
+        iterateReverse: jest.fn(),
+        makeLeader: jest.fn(),
+        makeFollower: jest.fn(),
+    };
+});
+
+jest.mock('../src/lib/rpc.js', () => {
+    return {
+        boardSlottingMarshaller: jest.fn(),
+        networkConfig: { rpcAddrs: ["http://127.0.0.1:26657"], chainName: "agoriclocal" },
+        makeAgoricNames: jest.fn()
+    };
+});
+
+jest.mock('../src/lib/wallet.js', () => {
+    return {
+        getCurrent: jest.fn(),
+    };
+});
+
+jest.mock('../src/oracle/helper.js', () => {
+    const originalModule = jest.requireActual("../src/oracle/helper")
+    return {
+        ...originalModule,
+        readVStorage: jest.fn()
+    };
+});
+
+import { readJSONFile, readVStorage } from '../src/oracle/helper.js';
+import { getOffers, checkSubmissionForRound, queryPrice, queryRound, getOraclesInvitations } from '../src/oracle/middleware';
+import { iterateReverse, makeLeader, makeFollower } from '@agoric/casting';
+import { boardSlottingMarshaller, makeAgoricNames } from '../src/lib/rpc.js'
+import { getCurrent } from '../src/lib/wallet.js'
```

Comment:
Would prefer if imports are done before mocking.

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/22a246264d8b555ea88ba97bf1b4929a7c97d99b